test/nlptest.jl

```diff
@@ -3,216 +3,193 @@
 #  License, v. 2.0. If a copy of the MPL was not distributed with this
 #  file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-function runnonlineartests(algorithm, mip_solvers)
+function runnonlineartests(branch_cut, mip_solver, nlp_solver)
+    algorithm = branch_cut ? "BC" : "OA"
 
-facts("Sparse matrix bug test") do
-    m = Model(solver=PajaritoSolver(algorithm=algorithm,mip_solver=mip_solver,cont_solver=nlp_solver))
+    facts("Sparse matrix bug test") do
+        m = Model(solver=PajaritoSolver(branch_cut=branch_cut, solver_mip=mip_solver, solver_con=nlp_solver))
 
-    @variable(m, x >= 0, start = 1, Int)
-    @variable(m, y >= 0, start = 1)
+        @variable(m, x >= 0, start = 1, Int)
+        @variable(m, y >= 0, start = 1)
 
-    @objective(m, Min, -3x - y)
+        @objective(m, Min, -3x - y)
 
-    @constraint(m, 3x + 10 <= 20)
+        @constraint(m, 3x + 10 <= 20)
+        @NLconstraint(m, y^2 <= 10)
 
-    @NLconstraint(m, y^2 <= 10)
+        @fact solve(m) --> :Optimal
+    end
 
-    @fact solve(m) --> :Optimal
+    facts("Convex constraint with LB and UB test") do
+        m = Model(solver=PajaritoSolver(branch_cut=branch_cut, solver_mip=mip_solver, solver_con=nlp_solver))
 
-end
-
-
-facts("Convex constraint with LB and UB") do
-    m = Model(solver=PajaritoSolver(algorithm=algorithm,mip_solver=mip_solver,cont_solver=nlp_solver))
-
-    @variable(m, x >= 0, start = 1, Int)
```

Comment:
Try to make minimal changes to make things easier to review. It takes extra effort to figure out what happened to this test when actually not much changed.